tests/CMakeLists.txt

```diff
@@ -1,9 +1,11 @@
 find_package(GTest REQUIRED)
 
-add_executable(encoder_tests encoder_test.cpp)
+if(NOT DEFINED NO_TESTS OR NOT NO_TESTS)
```

Comment:
I took a quick stab at a yocto package for this, so we don't have to pull gtest in as a dep when building without tests, this `NO_TESTS` check should either include the `find_package(GTest` or be put in the top level cmakelists around the `enable_testing()` and `add_subdirectory(tests)`.